src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala

```diff
@@ -151,27 +172,54 @@ object ScalafixInterface {
           val scalafixArguments = ScalafixAPI
             .fetchAndClassloadInstance(
               scalafixScalaBinaryVersion,
-              scalafixCustomResolvers.asJava
+              toolClasspath.repositories.asJava
             )
             .newArguments()
             .withMainCallback(callback)
-          val printStream =
+
+          val printStream = Arg.PrintStream(
             new PrintStream(
               LoggingOutputStream(
                 logger,
                 Level.Info
               )
             )
-          new ScalafixInterface(scalafixArguments, Nil)
-            .withArgs(
-              Arg.PrintStream(printStream),
-              Arg.ToolClasspath(
-                Nil,
-                scalafixDependencies,
-                scalafixCustomResolvers
-              )
+          )
+
+          Some(
+            (
+              new ScalafixInterface(scalafixArguments).withArgs(printStream),
+              Nil
             )
-        }
-      )
-    })
+          )
+      }
+    )
+
+    // stamp the files that might change, to potentialy force invalidation
+    // of the corresponding cache entry
+    val currentStamps =
+      toolClasspath.mutableFiles.map(IO.getModifiedTimeOrZero)
+
+    val (toolClasspathInterface, _) = cache.compute(
+      (
+        scalafixScalaBinaryVersion,
+        Some(toolClasspath)
+      ),
+      {
+        case Some((_, oldStamps)) if (currentStamps == oldStamps) =>
+          // cache hit, don't update
+          None
+        case _ =>
+          // cache miss or stale stamps, resolve custom rules artifacts and classload them
+          Some(
+            (
+              buildinRulesInterface.withArgs(toolClasspath),
```

Comment:
by replacing the former value with this, we make the former interface, together with its toolClasspath classloader, eligible for garbage collection